### PR TITLE
Fixed #27622 -- Allowed test client to accept vendor tree JSON content types.

### PR DESCRIPTION
**Pull request by desecho**
https://code.djangoproject.com/ticket/27622

### CODE REVIEW

File: tests/test_client_regress/views.py [hunk 1]
```diff
@@ -109,6 +109,10 @@ def return_json_response(request):
     return JsonResponse({'key': 'value'})
 
 
+def return_json_vendor_response(request):
```
Comment (timgraham) [inline tests/test_client_regress/views.py]: You could modify the existing `return_json_response` view to extra `content_type` from `request.GET` to allow more tests without creating a separate view for each one.

File: AUTHORS [hunk 2]
```diff
@@ -62,6 +62,7 @@ answer newbie questions, and generally made Django that much better:
     andy@jadedplanet.net
     Anssi Kääriäinen <akaariai@gmail.com>
     ant9000@netwise.it
+    Anton Samarchyan <desecho@gmail.com>
```
Comment (timgraham) [inline AUTHORS]: Please don't add this is every one of your PRs. It'll be added (sorted correctly) as part of https://github.com/django/django/pull/7754.

File: django/test/client.py [hunk 3]
```diff
@@ -690,7 +690,10 @@ def logout(self):
 
     def _parse_json(self, response, **extra):
         if not hasattr(response, '_json'):
-            if 'application/json' not in response.get('Content-Type'):
+            # Use regex to support JSON Vendor Tree specification
+            # https://tools.ietf.org/html/rfc6838#section-3.2
+            json_content_type_re = re.compile(r'^application\/(vnd\..+\+)?json$')
```
Comment (timgraham) [inline django/test/client.py]: Would be better to put this as a module constant to avoid compiling it over and over.

**Comment by desecho** [comment]
I made the changes.

**Pull request merged**